src/Datagrid/DatagridMapper.php

```diff
@@ -57,10 +60,10 @@ public function add(
         ?string $type = null,
         array $filterOptions = [],
         ?string $fieldType = null,
-        ?array $fieldOptions = null,
+        array $fieldOptions = [],
         array $fieldDescriptionOptions = []
     ): self {
-        if (\is_array($fieldOptions)) {
+        if ([] !== $fieldOptions) {
```

Comment:
I don't think the check is necessary now.

Before, `[]` was an array so we set `$filterOptions['field_options'] = []`

Reply:
I was worried to change the previous behavior, but will try without the if

Reply:
Done

Reply:
So null was leading to required false and empty array to empty array ?
And now empty array lead to required false ?

I dont really know why there is this default behaviour of required false.
We may remove this default behavior ?

Reply:
Don't really know, maybe it is better to recover the old behavior and open an issue to make sure we check why is this needed or if it can be simplified

Reply:
I agree that maybe it's better to leave it as it was and open an issue.

Reply:
Ok ! Can you open the issue @jordisala1991 ?

Reply:
I will revert this code to accept null (old behavior, and create a new issue)

Reply:
Done for the code, and for the issue: https://github.com/sonata-project/SonataAdminBundle/issues/6272